File: VisualBasicAssignment/screen/sai_keat/widget/payment/PaymentMethods.vb
Imports Guna.UI2.WinForms
Imports System.Data.Entity.Migrations

Public Class PaymentMethods

    Public master As Master
    Private Sub radVisa_CheckedChanged(sender As Object, e As EventArgs) Handles radVisa.CheckedChanged
        Dim btn As Guna2RadioButton = CType(sender, Guna2RadioButton)
        If (btn.Checked) Then
            radMaster.Checked = False
            radVisa.Checked = True
        End If
    End Sub

    Private Sub radMaster_CheckedChanged(sender As Object, e As EventArgs) Handles radMaster.CheckedChanged
        Dim btn As Guna2RadioButton = CType(sender, Guna2RadioButton)
        If (btn.Checked) Then
            radMaster.Checked = True
            radVisa.Checked = False
        End If
    End Sub

    Private Sub Guna2Button3_Click(sender As Object, e As EventArgs) Handles Guna2Button3.Click

        err.Tag = Nothing

        If Not ValidatePayment.isName(Me.txbName.Text) Then
            err.SetError(txbName, "Please type your name!")
            err.Tag = If(err.Tag, txbName)
        End If
        If radMaster.Checked Then
            If Not ValidatePayment.isMastercard(Me.txbCardNumber.Text) Then
                err.SetError(txbCardNumber, "Not master card format *Card number 16 number withoud white space!")
                err.Tag = If(err.Tag, txbCardNumber)
            End If
        Else
            If Not ValidatePayment.isVisacard(Me.txbCardNumber.Text) Then
                err.SetError(txbCardNumber, "Mot visa card format, Card number 16 number withoud white space!")
                err.Tag = If(err.Tag, txbCardNumber)
            End If
        End If
        If Not ValidatePayment.isCCV(Me.txbCCV.Text) Then
            err.SetError(txbCCV, "Please type your CCV!")
            err.Tag = If(err.Tag, txbCCV)
        End If
        If Not ValidatePayment.isExp(Me.txbExp.Text) Then
            err.SetError(txbExp, "Please type your EXP!")
            err.Tag = If(err.Tag, txbExp)
        End If

        If err.Tag Is Nothing Then
            Using db As FoodShopEntities1 = New FoodShopEntities1()
                Try
                    Dim pay As Payment = DatabaseConnections.GetCurrentPayment()
                    pay.customer_paid = pay.total_pay
                    pay.payment_meta = txbName.Text + "," + txbCardNumber.Text + "," + txbCCV.Text + "," + txbExp.Text
                    ' Set the payment status to completed
                    pay.payment_status = Guid.Parse("c595596d-8980-4138-bc20-a91056e1b1ce")
                    db.Payments.AddOrUpdate(pay)
                    db.SaveChanges()
                    master.Hide()

                    PaymentStep3.Show()
                Catch ex As Exception
                    Console.WriteLine("Exception caught: {0}", ex)
                End Try
            End Using
        End If

    End Sub
End Class

File: VisualBasicAssignment/screen/sai_keat/widget/payment/PaymentMethods.cs
using System;
using System.Data.Entity.Migrations;
using Guna.UI2.WinForms;

namespace VisualBasicAssignment
{
    public partial class PaymentMethods
    {
        public PaymentMethods()
        {
            InitializeComponent();
            _Guna2Button3.Name = "Guna2Button3";
            _radMaster.Name = "radMaster";
            _radVisa.Name = "radVisa";
        }

        public Master master;

        private void radVisa_CheckedChanged(object sender, EventArgs e)
        {
            Guna2RadioButton btn = (Guna2RadioButton)sender;
            if (btn.Checked)
            {
                radMaster.Checked = false;
                radVisa.Checked = true;
            }
        }

        private void radMaster_CheckedChanged(object sender, EventArgs e)
        {
            Guna2RadioButton btn = (Guna2RadioButton)sender;
            if (btn.Checked)
            {
                radMaster.Checked = true;
                radVisa.Checked = false;
            }
        }

        private void Guna2Button3_Click(object sender, EventArgs e)
        {
            err.Tag = null;
            if (!ValidatePayment.isName(txbName.Text))
            {
                err.SetError(txbName, "Please type your name!");
                err.Tag = err.Tag ?? txbName;
            }

            if (radMaster.Checked)
            {
                if (!ValidatePayment.isMastercard(txbCardNumber.Text))
                {
                    err.SetError(txbCardNumber, "Not master card format *Card number 16 number withoud white space!");
                    err.Tag = err.Tag ?? txbCardNumber;
                }
            }
            else if (!ValidatePayment.isVisacard(txbCardNumber.Text))
            {
                err.SetError(txbCardNumber, "Mot visa card format, Card number 16 number withoud white space!");
                err.Tag = err.Tag ?? txbCardNumber;
            }

            if (!ValidatePayment.isCCV(txbCCV.Text))
            {
                err.SetError(txbCCV, "Please type your CCV!");
                err.Tag = err.Tag ?? txbCCV;
            }

            if (!ValidatePayment.isExp(txbExp.Text))
            {
                err.SetError(txbExp, "Please type your EXP!");
                err.Tag = err.Tag ?? txbExp;
            }

            if (err.Tag is null)
            {
                using (var db = new FoodShopEntities1())
                {
                    try
                    {
                        var pay = DatabaseConnections.GetCurrentPayment();
                        pay.customer_paid = pay.total_pay;
                        pay.payment_meta = txbName.Text + "," + txbCardNumber.Text + "," + txbCCV.Text + "," + txbExp.Text;
                        // Set the payment status to completed
                        pay.payment_status = Guid.Parse("c595596d-8980-4138-bc20-a91056e1b1ce");
                        db.Payments.AddOrUpdate(pay);
                        db.SaveChanges();
                        master.Hide();
                        My.MyProject.Forms.PaymentStep3.Show();
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine("Exception caught: {0}", ex);
                    }
                }
            }
        }
    }
}